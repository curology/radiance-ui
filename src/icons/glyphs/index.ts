// @create-index

export { default as AcneBodySprayGlyph } from './acneBodySprayGlyph';
export { default as AcneBodyWashGlyph } from './acneBodyWashGlyph';
export { default as AcneFiveGlyph } from './acneFiveGlyph';
export { default as AcneGlyph } from './acneGlyph';
export { default as AcneOneGlyph } from './acneOneGlyph';
export { default as AcneSevenGlyph } from './acneSevenGlyph';
export { default as AcneThreeGlyph } from './acneThreeGlyph';
export { default as AhaBhaTonerSolutionGlyph } from './ahaBhaTonerSolutionGlyph';
export { default as AntiAgingAndAcneGlyph } from './antiAgingAndAcneGlyph';
export { default as AntiAgingGlyph } from './antiAgingGlyph';
export { default as AzelaicAcidGlyph } from './azelaicAcidGlyph';
export { default as BodySunscreenGlyph } from './bodySunscreenGlyph';
export { default as BreakGlyph } from './breakGlyph';
export { default as CheckBadgeGlyph } from './checkBadgeGlyph';
export { default as ChestGlyph } from './chestGlyph';
export { default as CleanserGlyph } from './cleanserGlyph';
export { default as ClindamycinGlyph } from './clindamycinGlyph';
export { default as ClockAlwaysGlyph } from './clockAlwaysGlyph';
export { default as ClockMostOfTheTimeGlyph } from './clockMostOfTheTimeGlyph';
export { default as ClockNeverGlyph } from './clockNeverGlyph';
export { default as ClockSometimesGlyph } from './clockSometimesGlyph';
export { default as CloggedPoresGlyph } from './cloggedPoresGlyph';
export { default as CombinationGlyph } from './combinationGlyph';
export { default as ConsultHistoryGlyph } from './consultHistoryGlyph';
export { default as CourseGlyph } from './courseGlyph';
export { default as CycleGlyph } from './cycleGlyph';
export { default as DarkSpotFormulaGlyph } from './darkSpotFormulaGlyph';
export { default as DarkSpotsGlyph } from './darkSpotsGlyph';
export { default as DesonideGlyph } from './desonideGlyph';
export { default as DexpanthenolGlyph } from './dexpanthenolGlyph';
export { default as DropperGlyph } from './dropperGlyph';
export { default as ExerciseGlyph } from './exerciseGlyph';
export { default as ExfoliatingMaskGlyph } from './exfoliatingMaskGlyph';
export { default as EyecreamGlyph } from './eyecreamGlyph';
export { default as FaceGlyph } from './faceGlyph';
export { default as FemaleGlyph } from './femaleGlyph';
export { default as FineLinesGlyph } from './fineLinesGlyph';
export { default as FirmnessGlyph } from './firmnessGlyph';
export { default as FlagCanadaGlyph } from './flagCanadaGlyph';
export { default as FlagUkGlyph } from './flagUKGlyph';
export { default as FlagUsaGlyph } from './flagUSAGlyph';
export { default as FormulaChangeGlyph } from './formulaChangeGlyph';
export { default as GlobeGlyph } from './globeGlyph';
export { default as GreenTeaGlyph } from './greenTeaGlyph';
export { default as HandWithCreamGlyph } from './handWithCreamGlyph';
export { default as HydratingMaskGlyph } from './hydratingMaskGlyph';
export { default as HydrocolloidBandageGlyph } from './hydrocolloidBandageGlyph';
export { default as HydrocortisoneGlyph } from './hydrocortisoneGlyph';
export { default as HydroquinoneCycleGlyph } from './hydroquinoneCycleGlyph';
export { default as HydroquinoneGlyph } from './hydroquinoneGlyph';
export { default as IntersexGlyph } from './intersexGlyph';
export { default as KojicAcidGlyph } from './kojicAcidGlyph';
export { default as LightbulbGlyph } from './lightbulbGlyph';
export { default as LipBalmGlyph } from './lipBalmGlyph';
export { default as LockGlyph } from './lockGlyph';
export { default as LunchGlyph } from './lunchGlyph';
export { default as MaleGlyph } from './maleGlyph';
export { default as MelasmaGlyph } from './melasmaGlyph';
export { default as MessageGlyph } from './messageGlyph';
export { default as MetronidazoleGlyph } from './metronidazoleGlyph';
export { default as MicellarWaterGlyph } from './micellarWaterGlyph';
export { default as MixIngredientsGlyph } from './mixIngredientsGlyph';
export { default as MoisturizerGlyph } from './moisturizerGlyph';
export { default as MoonGlyph } from './moonGlyph';
export { default as MortorGlyph } from './mortorGlyph';
export { default as NeckGlyph } from './neckGlyph';
export { default as NiacinamideGlyph } from './niacinamideGlyph';
export { default as NoMakeupGlyph } from './noMakeupGlyph';
export { default as NoneGlyph } from './noneGlyph';
export { default as OftenDryGlyph } from './oftenDryGlyph';
export { default as OftenOilyGlyph } from './oftenOilyGlyph';
export { default as OilCleanserGlyph } from './oilCleanserGlyph';
export { default as OtherGlyph } from './otherGlyph';
export { default as PackagingGlyph } from './packagingGlyph';
export { default as PalmTreeGlyph } from './palmTreeGlyph';
export { default as PillsGlyph } from './pillsGlyph';
export { default as RelaxGlyph } from './relaxGlyph';
export { default as ResveratrolGlyph } from './resveratrolGlyph';
export { default as RichMoisturizerGlyph } from './richMoisturizerGlyph';
export { default as RocketshipGlyph } from './rocketshipGlyph';
export { default as SelfieGlyph } from './selfieGlyph';
export { default as SleepGlyph } from './sleepGlyph';
export { default as StressGlyph } from './stressGlyph';
export { default as SunGlyph } from './sunGlyph';
export { default as SunMoonGlyph } from './sunMoonGlyph';
export { default as SunblockGlyph } from './sunblockGlyph';
export { default as SunscreenGlyph } from './sunscreenGlyph';
export { default as SuperbottleGlyph } from './superbottleGlyph';
export { default as SupportCenterGlyph } from './supportCenterGlyph';
export { default as TextureGlyph } from './textureGlyph';
export { default as TimeAlwaysGlyph } from './timeAlwaysGlyph';
export { default as TimeMostOfTheTimeGlyph } from './timeMostOfTheTimeGlyph';
export { default as TimeNeverGlyph } from './timeNeverGlyph';
export { default as TimeSometimesGlyph } from './timeSometimesGlyph';
export { default as TranexamicAcidGlyph } from './tranexamicAcidGlyph';
export { default as TretinoinGlyph } from './tretinoinGlyph';
export { default as VeryDryGlyph } from './veryDryGlyph';
export { default as VeryOilyGlyph } from './veryOilyGlyph';
export { default as VitaminCGlyph } from './vitaminCGlyph';
export { default as VitaminCSerumGlyph } from './vitaminCSerumGlyph';
export { default as WashFaceGlyph } from './washFaceGlyph';
export { default as WrinklesGlyph } from './wrinklesGlyph';
export { default as YesGlyph } from './yesGlyph';
export { default as ZincGlyph } from './zincGlyph';
export { default as ZincSoapBarGlyph } from './zincSoapBarGlyph';
