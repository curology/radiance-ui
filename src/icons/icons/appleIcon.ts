import { AppleIcon } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) => useIcon(AppleIcon, AppleIcon, props);
