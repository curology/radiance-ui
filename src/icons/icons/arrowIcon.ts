import { ArrowIcon } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) => useIcon(ArrowIcon, ArrowIcon, props);
