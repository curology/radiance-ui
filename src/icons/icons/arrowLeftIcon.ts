import { ArrowLeftIcon, ArrowLeftIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) =>
  useIcon(ArrowLeftIcon, ArrowLeftIconSecondary, props);
