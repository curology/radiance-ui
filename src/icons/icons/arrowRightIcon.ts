import { ArrowRightIcon, ArrowRightIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) =>
  useIcon(ArrowRightIcon, ArrowRightIconSecondary, props);
