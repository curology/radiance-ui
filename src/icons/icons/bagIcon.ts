import { BagIcon, BagIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) => useIcon(BagIcon, BagIconSecondary, props);
