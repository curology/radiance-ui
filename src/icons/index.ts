/*
 * Note: this file has to be manually updated if we add more types of icons.
 * We do not generate it via create-index
 */

export * from './emojis';
export * from './glyphs';
export * from './icons';
export * from './logos';
export * from './navIcons';
