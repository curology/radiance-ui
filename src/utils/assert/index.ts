import { isDefined } from '../isDefined';

function assert(value: boolean, message?: string): asserts value;
function assert<T>(
  value: T | null | undefined,
  message?: string,
): asserts value is T;
/**
 * @see https://gist.github.com/jamiebuilds/2a40f565cba317a80bed1eec127340f6
 */
function assert(value: unknown, message?: string) {
  if (value === false || !isDefined(value)) {
    throw new Error(message ?? 'Assertion failed');
  }
}

export { assert };
