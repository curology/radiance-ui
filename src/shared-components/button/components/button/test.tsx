import React from 'react';
import { userEvent, render } from 'src/tests/testingLibraryHelpers';
import { assert } from 'src/utils/assert';

import { CameraIcon } from '../../../../icons';

import { Button } from './index';

describe('<Button />', () => {
  /**
   * TODO: Fix Emotion 11 CI snapshot serializer order issue
   */
  // eslint-disable-next-line jest/no-disabled-tests
  describe.skip('UI snapshots', () => {
    it('renders with props', () => {
      const { container } = render(
        <Button disabled onClick={() => undefined} icon={<CameraIcon />}>
          Button Text
        </Button>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });
  });

  describe('onClick callback', () => {
    it('should be invoked onClick', () => {
      const spy = jest.fn();
      const { container } = render(<Button onClick={spy}>Button Text</Button>);

      assert(container.firstElementChild);
      userEvent.click(container.firstElementChild);

      expect(spy).toHaveBeenCalled();
    });

    it('should not be invoked if disabled', () => {
      const spy = jest.fn();
      const { container } = render(
        <Button disabled onClick={spy}>
          Button Text
        </Button>,
      );

      assert(container.firstElementChild);
      userEvent.click(container.firstElementChild);

      expect(spy).not.toHaveBeenCalled();
    });

    it('should not be invoked if loading', () => {
      const spy = jest.fn();
      const { container } = render(
        <Button isLoading onClick={spy}>
          Button Text
        </Button>,
      );

      assert(container.firstElementChild);
      userEvent.click(container.firstElementChild);

      expect(spy).not.toHaveBeenCalled();
    });
  });
});
