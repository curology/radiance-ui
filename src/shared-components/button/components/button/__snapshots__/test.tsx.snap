// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> UI snapshots renders with props 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0) scale(1, 1);
    -moz-transform: translate3d(0, 0, 0) scale(1, 1);
    -ms-transform: translate3d(0, 0, 0) scale(1, 1);
    transform: translate3d(0, 0, 0) scale(1, 1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px, 0, 0) scale(1, 1);
    -moz-transform: translate3d(12px, 0, 0) scale(1, 1);
    -ms-transform: translate3d(12px, 0, 0) scale(1, 1);
    transform: translate3d(12px, 0, 0) scale(1, 1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -moz-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -ms-transform: translate3d(16px, 0, 0) scale(.85, .85);
    transform: translate3d(16px, 0, 0) scale(.85, .85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0);
    -moz-transform: translate3d(16px, 0, 0);
    -ms-transform: translate3d(16px, 0, 0);
    transform: translate3d(16px, 0, 0);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0) scale(1, 1);
    -moz-transform: translate3d(0, 0, 0) scale(1, 1);
    -ms-transform: translate3d(0, 0, 0) scale(1, 1);
    transform: translate3d(0, 0, 0) scale(1, 1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px, 0, 0) scale(1, 1);
    -moz-transform: translate3d(12px, 0, 0) scale(1, 1);
    -ms-transform: translate3d(12px, 0, 0) scale(1, 1);
    transform: translate3d(12px, 0, 0) scale(1, 1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -moz-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -ms-transform: translate3d(16px, 0, 0) scale(.85, .85);
    transform: translate3d(16px, 0, 0) scale(.85, .85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0);
    -moz-transform: translate3d(16px, 0, 0);
    -ms-transform: translate3d(16px, 0, 0);
    transform: translate3d(16px, 0, 0);
  }
}

@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0) scale(1, 1);
    -moz-transform: translate3d(0, 0, 0) scale(1, 1);
    -ms-transform: translate3d(0, 0, 0) scale(1, 1);
    transform: translate3d(0, 0, 0) scale(1, 1);
  }

  20% {
    opacity: .95;
    -webkit-transform: translate3d(12px, 0, 0) scale(1, 1);
    -moz-transform: translate3d(12px, 0, 0) scale(1, 1);
    -ms-transform: translate3d(12px, 0, 0) scale(1, 1);
    transform: translate3d(12px, 0, 0) scale(1, 1);
  }

  35% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -moz-transform: translate3d(16px, 0, 0) scale(.85, .85);
    -ms-transform: translate3d(16px, 0, 0) scale(.85, .85);
    transform: translate3d(16px, 0, 0) scale(.85, .85);
  }

  100% {
    opacity: 0;
    -webkit-transform: translate3d(16px, 0, 0);
    -moz-transform: translate3d(16px, 0, 0);
    -ms-transform: translate3d(16px, 0, 0);
    transform: translate3d(16px, 0, 0);
  }
}

.emotion-0 {
  color: #524D6E;
  font-size: 0.75rem;
  line-height: 1.67;
  font-weight: bold;
  letter-spacing: 1px;
  text-transform: uppercase;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: block;
  margin: 0;
  min-height: 52px;
  opacity: 1;
  padding: 0 1.5rem;
  position: relative;
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #EDEDF0;
  border-color: #EDEDF0;
  color: #C3C0CD;
  cursor: not-allowed;
  fill: #C3C0CD;
  min-width: 208px;
  max-width: 325px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.emotion-0:hover {
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
}

.emotion-0:active,
.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 1rem 0;
  -webkit-transition: -webkit-transform 350ms;
  transition: transform 350ms;
  width: 100%;
  -webkit-transform: translateX(0);
  -moz-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.emotion-2>svg {
  opacity: 1;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
  margin-right: 1rem;
  margin-top: -5px;
}

.emotion-4 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-5 {
  line-height: 1.5;
  margin: 0;
  padding-top: 2px;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  position: absolute;
  bottom: 0;
  right: 0;
  top: 0;
  margin-top: -6px;
  width: 38px;
  opacity: 0;
}

.emotion-7 span {
  border-radius: 50%;
  display: inline-block;
  height: 4px;
  width: 4px;
  position: absolute;
}

.emotion-7 span:nth-of-type(1) {
  -webkit-animation: animation-0 1750ms infinite linear;
  animation: animation-0 1750ms infinite linear;
}

.emotion-7 span:nth-of-type(2) {
  -webkit-animation: animation-0 1750ms infinite linear -300ms;
  animation: animation-0 1750ms infinite linear -300ms;
}

.emotion-7 span:nth-of-type(3) {
  -webkit-animation: animation-0 1750ms infinite linear -600ms;
  animation: animation-0 1750ms infinite linear -600ms;
}

<button
  class="emotion-0 emotion-1"
  disabled=""
  type="button"
>
  <div
    class="emotion-2 emotion-3"
  >
    <svg
      class="emotion-4"
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Camera Icon
      </title>
      <path
        d="M10.69 2.25L12.15 4.02H14.76V13.45H1.26V4.02H3.89L4.27 3.56L5.33 2.26H10.69V2.25ZM11.28 1H4.73L3.29 2.76H0V14.7H16V2.77H12.73L11.28 1Z"
        fill="currentColor"
      />
      <path
        d="M8.00999 6.3999C9.12999 6.3999 10.03 7.3099 10.03 8.4199C10.03 9.5299 9.11999 10.4399 8.00999 10.4399C6.89999 10.4399 5.98999 9.5299 5.98999 8.4199C5.98999 7.3099 6.88999 6.3999 8.00999 6.3999ZM8.00999 5.1499C6.19999 5.1499 4.73999 6.6199 4.73999 8.4199C4.73999 10.2199 6.20999 11.6899 8.00999 11.6899C9.81999 11.6899 11.28 10.2199 11.28 8.4199C11.28 6.6199 9.80999 5.1499 8.00999 5.1499Z"
        fill="currentColor"
      />
    </svg>
    <span
      class="emotion-5 emotion-6"
    >
      Button Text
    </span>
  </div>
  <div
    class=" emotion-7 emotion-8"
    disabled=""
  >
    <div>
      <span />
      <span />
      <span />
    </div>
  </div>
</button>
`;
